core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -746,6 +746,14 @@ public void failShard(String reason, Throwable e) {
         engine().failEngine(reason, e);
```

Comment:
can we use failShard(reason, true, e) here? will be clear what it does. Also change the java docs please.